src/session_impl.cpp

```diff
@@ -448,19 +447,15 @@ namespace aux {
 		TORRENT_ASSERT_VAL(!ec, ec);
 
 		update_time_now();
+		m_disk_thread.set_settings(&pack, m_alerts);
```

Comment:
is this the issue?

Reply:
previously this would be called by ``apply_settings_pack()``, which I removed (because it ended up being redundant when ``m_settings`` was already initialized to the same settings, from the constructor)